src/modules/admin/components/AdminReferralPostingForm.tsx

```diff
@@ -0,0 +1,100 @@
+import { useCallback, useMemo, useState } from 'react';
+
+import { emptyErrorState, partitionValidations } from '@/modules/errors/util';
+import DynamicForm, {
+  DynamicFormOnSubmit,
+} from '@/modules/form/components/DynamicForm';
+import DynamicView from '@/modules/form/components/viewable/DynamicView';
+import { AdminReferralPostingDefinition } from '@/modules/form/data';
+import useInitialFormValues from '@/modules/form/hooks/useInitialFormValues';
+import { SubmitFormAllowedTypes } from '@/modules/form/types';
+import { transformSubmitValues } from '@/modules/form/util/formUtil';
+import {
+  ReferralPostingDetailFieldsFragment,
+  ReferralPostingInput,
+  useUpdateReferralPostingMutation,
+} from '@/types/gqlTypes';
+
+interface Props {
+  referralPosting: ReferralPostingDetailFieldsFragment;
+  readOnly?: boolean;
+}
+
+const AdminReferralPostingForm: React.FC<Props> = ({
+  readOnly = false,
+  ...props
+}) => {
+  const referralPosting = useMemo(() => {
+    const record = props.referralPosting;
+    // the status is denied pending status but we don't allow this option in the form. Set it to null instead
+    return {
+      ...record,
+      status: record.status == 'denied_pending_status' ? null : record.status,
+    };
+  }, [props.referralPosting]);
+
+  const formDefinition = AdminReferralPostingDefinition;
+  const [errors, setErrors] = useState(emptyErrorState);
+  const [mutate, { loading }] = useUpdateReferralPostingMutation({
+    onCompleted: (data) => {
+      const { errors: remoteErrors = [] } = data.updateReferralPosting || {};
+      if (remoteErrors.length) {
+        setErrors(partitionValidations(remoteErrors));
+      }
+    },
+    onError: (apolloError) => setErrors({ ...emptyErrorState, apolloError }),
+  });
+  const handleSubmit: DynamicFormOnSubmit = useCallback(
+    ({ values }) => {
+      const input = transformSubmitValues({
+        definition: formDefinition,
+        values,
+        keyByFieldName: true,
+      }) as ReferralPostingInput;
+
+      mutate({
+        variables: {
+          id: referralPosting.id,
+          input,
+        },
+      });
+    },
+    [mutate, referralPosting.id, formDefinition]
+  );
+
+  const initialValues = useInitialFormValues({
+    definition: formDefinition,
+    record: referralPosting as unknown as SubmitFormAllowedTypes,
+    localConstants: { hasReferralRequest: !!referralPosting.postingIdentifier },
```

Comment:
There is an ability to "admin assign" from the external system, whereby a new referral is sent to the HMIS even though it is not "fulfilling" any vacancy request. In that case, the Referral and Posting would both have Identifiers, but they would not be linked to a ReferralRequest.

In that scenario, we probably shouldn't show the "re-request" option. Sending a referral request would fail because there is no identifier there. My bad, that detail was not included in the ticket.


```
    localConstants: { hasReferralRequest: !!referralPosting.postingIdentifier && !!referralPosting.referralRequest },
```

Reply:
got it, I adjusted the condition and query to include the referral request